stan/math/rev/core/operator_plus_equal.hpp

```diff
@@ -9,7 +9,7 @@ namespace stan {
 namespace math {
 
 inline var& var::operator+=(var b) {
-  vi_ = new internal::add_vv_vari(vi_, b.vi_);
+  vi_ = new internal::add_vari<vari*, vari*>(vi_, b.vi_);
```

Comment:
Yes, if they're always pointes, then the template parameter can just be the type without the pointer.

But when is the type not going to be `vari` for `add_vari`?

Reply:
Oh, I see the example on line 21.  This is going to store everything.  This seems like it could get very confusing for the code for the functions.

Reply:
For `add_vari<vari*, double>`. Since we want the double to be the actual type here and not a pointer it could be better to specify this at the call site. Otherwise in the code we'll have to have some wrapper around the signatures to make them a pointer or not for vari which I think would be weird